migration/workspace-query-separation/src/main/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigration.java

```diff
@@ -0,0 +1,205 @@
+        .map(Result::getMetacard)
+        .filter(Objects::nonNull)
+        .forEach(this::migrateWorkspaceMetacard);
+
+    LOGGER.debug("Completed workspace query data migration");
+    System.out.println("Completed: Workspace Query Data Migration");
+  }
+
+  private void migrateWorkspaceMetacard(Metacard workspaceMetacard) {
+    LOGGER.debug("Migrating workspace metacard with id [{}]", workspaceMetacard.getId());
+
+    final Attribute queriesAttribute = workspaceMetacard.getAttribute(WORKSPACE_QUERIES);
+    final List<Serializable> queriesAsXml = queriesAttribute.getValues();
+
+    LOGGER.debug(
+        "Beginning migration of {} query metacards from workspace [{}]",
+        queriesAsXml.size(),
+        workspaceMetacard.getId());
+
+    List<Metacard> queryMetacards =
+        queriesAsXml
+            .stream()
+            .filter(String.class::isInstance)
+            .map(String.class::cast)
+            .map(xml -> xml.getBytes(StandardCharsets.UTF_8))
+            .map(ByteArrayInputStream::new)
+            .map(this::xmlToMetacard)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+    LOGGER.debug("Creating query metacards for workspace [{}]", workspaceMetacard.getId());
+
+    CreateResponse response = createMetacards(queryMetacards);
+
+    if (response != null) {
+      List<String> createdMetacardIds =
+          response.getCreatedMetacards().stream().map(Metacard::getId).collect(Collectors.toList());
+
+      LOGGER.debug("Created query metacards with IDs {}", createdMetacardIds);
+
+      workspaceMetacard.setAttribute(
+          new AttributeImpl(WORKSPACE_QUERIES, (Serializable) createdMetacardIds));
+
+      updateMetacard(workspaceMetacard);
+
+      LOGGER.debug(
+          "Updated workspace metacard [{}] with query metacard associations {}",
+          workspaceMetacard.getId(),
+          createdMetacardIds);
+    } else {
+      LOGGER.debug("An error occurred while creating query metacards");
+    }
+  }
+
+  private UpdateResponse updateMetacard(Metacard metacard) {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<UpdateResponse>)
+            () -> {
+              try {
+                return catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+              } catch (IngestException | SourceUnavailableException e) {
+                LOGGER.debug("Error updating workspace metacard [{}]", metacard.getId(), e);
+              }
+              return null;
+            });
+  }
+
+  private CreateResponse createMetacards(List<Metacard> metacards) {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<CreateResponse>)
+            () -> {
+              try {
+                return catalogFramework.create(new CreateRequestImpl(metacards));
+              } catch (IngestException | SourceUnavailableException e) {
+                LOGGER.debug("Error while creating query metacards {}", metacards, e);
```

Comment:
Neither `QueryMetacardImpl` nor `MetacardImpl` override `toString()` so I don't think printing the metacard list will provide any useful information.